src/main/java/com/trustev/web/ApiClient.java

```diff
@@ -22,727 +21,1339 @@
-		}
-		
-		public Date getExpireAt() {
-			return expireAt;
-		}
-		
-		@JsonProperty("ExpireAt")
-		public void setExpireAt(String expireAt) throws ParseException {
-			String expireTimeStamp = expireAt.substring(0, expireAt.lastIndexOf(".")+ 1 ) + "000Z";
-			DateFormat m_ISO8601Local = new SimpleDateFormat ("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-			m_ISO8601Local.setTimeZone(TimeZone.getTimeZone("UTC"));
-			this.expireAt = m_ISO8601Local.parse(expireTimeStamp);
-		}
-		
-		public String getCredentialType() {
-			return credentialType;
-		}
-		
-		@JsonProperty("CredentialType")
-		public void setCredentialType(String credentialType) {
-			this.credentialType = credentialType;
-		}
-		
-		private String apiToken;
-		private Date expireAt;
-		private String credentialType;
-	}
-	
+    private static Map<String, MerchantSite> merchantSites;
+
+    static {
+        merchantSites = new HashMap<String, MerchantSite>();
+    }
+
+    /**
+     * Initialize the Trustev class by passing in your UserName, Secret and Password.
+     * If you do not have this information, please contact our Integration Team - integrate@trustev.com
+     *
+     * @param userName Your Trustev Username
+     * @param password Your Trustev Password
+     * @param secret   Your Trustev Secret
+     * @param baseUrl  The base url for the api calls (either US or EU)
+     */
+    public static void SetUp(String userName, String password, String secret, BaseUrl baseUrl) {
+        if (!hasMerchantSite(userName)) {
+
+            String url = "";
+
+            //Ensure that the baseUrl which the client provides is either US or EU
+            if (baseUrl.equals(BaseUrl.US)) {
+                url = "https://app.trustev.com/api/v2.0";
+            } else {
+                url = "https://app-eu.trustev.com/api/v2.0";
+            }
+
+            merchantSites.put(userName, new MerchantSite(userName, password, secret, url));
+        }
+
+    }
+
+    /**
+     * Initialize the Trustev class by passing in your UserName, Secret and Password.
+     * If you do not have this information, please contact our Integration Team - integrate@trustev.com
+     *
+     * @param userName Your Trustev Username
+     * @param password Your Trustev Password
+     * @param secret   Your Trustev Secret
+     * @param baseUrl  A url string specifying base url for the api calls
+     */
+    public static void SetUp(String userName, String password, String secret, String baseUrl) {
+        merchantSites.put(userName, new MerchantSite(userName, password, secret, baseUrl));
+    }
+
+    /**
+     * Check if the Merchant Site associated with the username is already in the Map of merchantSites
+     *
+     * @param userName the username for the given merchant site
+     * @return true if the merchant site is already registered, false if the merchant site is not registered (or if the map of merchantSites is empty)
+     */
+    public static boolean hasMerchantSite(String userName) {
```

Comment:
Should some of these merchantSite methods be made private?? Do we want to expose all this functionality??